Assets/Scripts/Army/Army.cs

```diff
@@ -433,25 +445,86 @@ private void CmdAttack(Entity entity)
             Debug.LogError("Entity has no health component");
             return;
         }
-        // Debug.Log("Attacking");
         attackTarget = entity;
         attackTarget.GetComponent<EntityHealth>().OnDie.AddListener(HandleAttackTargetOnDie);
         SetState(ArmyState.Attacking);
     }
+
+    /// <summary>
+    /// Called in the update loop when the army is attacking
+    /// </summary>
+    [Server]
+    private void Attack()
+    {
+        if (attackTarget == null) // Target died
+        {
+            // Debug.Log("Target died");
+            state = ArmyState.Idle;
+        }
+        else
+        {
+            if (Vector3.Distance(transform.position, attackTarget.transform.position) <= attackRange)
+            {
+                entityMovement.Stop();
```

Comment:
After playtesting, we can decide if we should add a separate chase range and attack range to not have attack constantly interrupted